src/main/scala/org/broadinstitute/hail/expr/package.scala

```diff
@@ -3,7 +3,7 @@ package org.broadinstitute.hail
 package object expr {
   type SymbolTable = Map[String, (Int, BaseType)]
 
-  type Aggregator = (() => Any, (Any) => Any, (Any, Any) => Any, Int)
+  type Aggregator = (() => Any, (Any, Any) => Any, (Any, Any) => Any, Int)
```

Comment:
I think this should be a typed case class.  How about:

```
type Aggregator = TypedAggregator[Any, Any]
abstract class TypedAggregator[T, S] { self =>
  def zero: S
  def seqOp(x: T, acc: S): S
  def combOp...
  def erase: Aggregator = new Aggregator[Any, Any] {
    def zero = self.zero
    def seqOp(x: Any, acc: Any) = self.seqOp(x.asInstanceOf[T], acc.asInstanceOf[S])
    ...
  }
}
```

Then below when you create aggregators, you can just say `new TypedAggregator[T, S] { def zero = ... }`.